### PR TITLE
Document staging folder

### PR DESCRIPTION
**Pull request by ptomato**
This intends to make explicit in the contributor and interpreter docs, the intention of the staging folder and how implementations should handle it in their test262 interpreters.

Also adds a note about the `test/intl402/` folder since there wasn't one previously.

Closes: #3649

### CODE REVIEW

**Review by gibson042 — APPROVED** [review]

**Review by linusg — APPROVED** [review]
Perfect, thanks!

File: CONTRIBUTING.md
Comment (ljharb) [inline CONTRIBUTING.md:274]: i'd prefer to avoid hard-wrapping, if possible, but not a blocker.

**Review by ljharb — DISMISSED** [review]
LGTM, once @syg approves

File: CONTRIBUTING.md
Comment (Ms2ger) [inline CONTRIBUTING.md:264]: Consider clarifying "late-stage"

**Review by Ms2ger — APPROVED** [review]

File: CONTRIBUTING.md
Comment (ptomato) [reply CONTRIBUTING.md]: I don't mind changing this, but it should really be a lint rule. The existing text in the files is a mix of wrapping styles, and you can't see the difference when it's rendered anyway. I think we should pick a style ([here is a comprehensive blog post about pros and cons of different wrapping styles](https://mtsknn.fi/blog/4-1-wrapping-styles-for-markdown-prose-and-code-comments/)) and just automate it.

File: CONTRIBUTING.md
Comment (ptomato) [reply CONTRIBUTING.md]: I realized "late-stage" is no different from the rule about what tests are [accepted](https://github.com/tc39/test262/blob/main/CONTRIBUTING.md#acceptable-tests) in the main tree, so I rephrased this.

**Comment by ptomato** [comment]
Updated. I've also added a paragraph about implementations being able to designate someone(s) who can land tests in staging independently.

**Review by syg — APPROVED** [review]
This is great, thanks for documenting!

File: CONTRIBUTING.md
Comment (ljharb) [reply CONTRIBUTING.md]: Totally fine with doing that (i prefer a sentence per line, since "semantic linebreaks" isn't automateable)

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**
